iothub/device/tests/Authentication/DeviceClientConnectionStringExceptionTests.cs

```diff
@@ -37,7 +37,7 @@ public void DeviceClientConnectionStringDefaultScopeDefaultCredentialTypeMissing
         }
 
         [TestMethod]
-        [ExpectedException(typeof(InvalidOperationException))]
+        [ExpectedException(typeof(ArgumentException))]
```

Comment:
These tests are now back to asserting an ArgumentException

Reply:
Note on why this changed even though we didn't change the exception type in the previous PR (#2625 ):
Previously, the SDK would first throw an InvalidOperationException, and then, in trying to throw the exception, would do a ".ToString()" on the class (IotHUbConnectionStringBuilder). The "ToString()" method here would again throw an ArgumentException, which was returned to the caller.

This flow is now updated to throw an ArgumentException right from the beginning.